cmd/influxd/inspect/tsm_verify_test.go

```diff
@@ -0,0 +1,120 @@
+package inspect
+
+import (
+	"bytes"
+	"encoding/binary"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/influxdata/influxdb/v2/tsdb/engine/tsm1"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInvalidChecksum(t *testing.T) {
+	path := newChecksumTest(t, true)
+	defer os.RemoveAll(path)
+
+	verify := NewTSMVerifyCommand()
+	verify.SetOut(bytes.NewBufferString(""))
+	verify.SetArgs([]string{"--dir", path})
+	require.Error(t, verify.Execute())
+}
+
+func TestValidChecksum(t *testing.T) {
+	path := newChecksumTest(t, false)
+	defer os.RemoveAll(path)
+
+	verify := NewTSMVerifyCommand()
+	b := bytes.NewBufferString("")
+	verify.SetOut(b)
+	verify.SetArgs([]string{"--dir", path})
+	require.NoError(t, verify.Execute())
+
+	out, err := ioutil.ReadAll(b)
+	require.NoError(t, err)
+	require.True(t, strings.Contains(string(out), "Broken Blocks: 0 / 1"))
+}
+
+func TestInvalidUTF8(t *testing.T) {
+	path := newUTFTest(t, true)
+	defer os.RemoveAll(path)
+
+	verify := NewTSMVerifyCommand()
+	verify.SetOut(bytes.NewBufferString(""))
+	verify.SetArgs([]string{"--dir", path, "--check-utf8"})
+	require.Error(t, verify.Execute())
+}
+
+func TestValidUTF8(t *testing.T) {
+	path := newUTFTest(t, false)
+	defer os.RemoveAll(path)
+
+	verify := NewTSMVerifyCommand()
+	b := bytes.NewBufferString("")
+	verify.SetOut(b)
+	verify.SetArgs([]string{"--dir", path, "--check-utf8"})
+	require.NoError(t, verify.Execute())
+
+	out, err := ioutil.ReadAll(b)
+	require.NoError(t, err)
+	require.True(t, strings.Contains(string(out), "Invalid Keys: 0 / 1"))
+}
+
+func newUTFTest(t *testing.T, withError bool) string {
+	t.Helper()
+
+	dir, err := ioutil.TempDir("", "verify-tsm")
+	require.NoError(t, err)
+
+	f, err := ioutil.TempFile(dir, "verifytsmtest*"+"."+tsm1.TSMFileExtension)
+	require.NoError(t, err)
+
+	w, err := tsm1.NewTSMWriter(f)
+	require.NoError(t, err)
+	defer w.Close()
+
+	values := []tsm1.Value{tsm1.NewValue(0, 1.0)}
+	require.NoError(t, w.Write([]byte("cpu"), values))
+
+	if withError {
+		require.NoError(t, binary.Write(f, binary.BigEndian, []byte("foobar\n")))
+	}
+
+	require.NoError(t, w.WriteIndex())
+
+	return dir
+}
+
+func newChecksumTest(t *testing.T, withError bool) string {
+	t.Helper()
+
+	dir, err := ioutil.TempDir("", "verify-tsm")
+	require.NoError(t, err)
+
+	f, err := ioutil.TempFile(dir, "verifytsmtest*"+"."+tsm1.TSMFileExtension)
+	require.NoError(t, err)
+
+	w, err := tsm1.NewTSMWriter(f)
+	require.NoError(t, err)
+	defer w.Close()
+
+	values := []tsm1.Value{tsm1.NewValue(0, "entry")}
+	require.NoError(t, w.Write([]byte("cpu"), values))
+
+	require.NoError(t, w.WriteIndex())
+
+	if withError {
+		fh, err := os.OpenFile(f.Name(), os.O_RDWR, 0)
+		require.NoError(t, err)
+		defer fh.Close()
+
+		_, err = fh.WriteAt([]byte("foobar"), 0)
+		require.NoError(t, err)
+
+		require.NoError(t, w.WriteIndex())
```

Comment:
I'd expect this 2nd call to fix up the checksums and avoid an error... what error message do you end up seeing with the code as-is?

Reply:
`Error: indirectIndex: not enough data for min time`